pytest_xvfb.py

```diff
@@ -130,8 +132,13 @@ def pytest_addoption(parser):
 
 
 def pytest_configure(config):
-    if config.getoption('--no-xvfb') or not xvfb_available():
+    if config.getoption('--no-xvfb') or not sys.platform.startswith('linux'):
         config.xvfb = None
+    elif (not xvfb_available() and sys.platform.startswith('linux') and
+          os.environ.get('DISPLAY') is not None):
+        config.xvfb = None
+        warnings.warn(
+            'pytest-xvfb could not find Xvfb. You can install it to avoid windows from opening.')
```

Comment:
This would try to start Xvfb with no `DISPLAY` set. Why not simply:

```python
if config.getoption('--no-xvfb') or not xvfb_available():
    config.xvfb = None
    if sys.platform.startswith('linux') and 'DISPLAY' in os.environ and not config.getoption('--no-xvfb') :
        # ...
```

Also, this is a Python warning and not a pytest warning, so the user isn't likely to see it.